### PR TITLE
WIP CFEP-13 validation and some various bug fixes

### PR DESCRIPTION
**Pull request by beckermr**
Enables the new CFEP-13 package validation. 

Checklist
* [x] Added a ``news`` entry

### CODE REVIEW

File: conda_smithy/templates/appveyor.yml.tmpl [hunk 1]
```diff
@@ -56,7 +56,8 @@ install:
 build: off
 
 test_script:
-    - conda.exe build {{ recipe_dir }} -m .ci_support\%CONFIG%.yaml
+    - cmd: conda.exe build {{ recipe_dir }} -m .ci_support\%CONFIG%.yaml
+    - cmd: validate_recipe_outputs
```
Comment (beckermr) [inline conda_smithy/templates/appveyor.yml.tmpl:60]: This will be added to the ci setup package.

**Comment by beckermr** [comment]
ok @conda-forge/core This PR is ready for review!

File: conda_smithy/configure_feedstock.py [hunk 2]
```diff
@@ -1385,7 +1385,9 @@ def _load_forge_config(forge_dir, exclusive_config_file):
         },
         "recipe_dir": "recipe",
         "skip_render": [],
-        "bot": {"automerge": False,},
+        "bot": {"automerge": False},
+        "conda_forge_output_validation": False,
+        "conda_forge_ci_setup_version": "2",
```
Comment (isuruf) [inline conda_smithy/configure_feedstock.py:1390]: This is too much customization. Let's just change 2 to 3

Comment (beckermr) [reply conda_smithy/configure_feedstock.py]: done.

**Comment by beckermr** [comment]
Superseded by #1295

**Pull request closed without merging**